### PR TITLE
Add interoperability endpoints for mainchain and sidechain - Closes #7225

### LINKED ISSUE #7225: Implement endpoints for Interoperability module
### Description

Implement endpoints for Interoperability module

Endpoints:
- getChainAccount(chainID)
- getChannel(chainID)
- getOwnChainAccount()
- getTerminatedStateAccount(chainID)
- getTerminatedOutboxAccount(chainID)

Module changes:
- Add these endpoints to metadata of Interoperability module.

### Acceptance Criteria

- Should have all the unit tests for all the endpoints

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #7225 

### How was it solved?

- Added endpoints in sidechain and mainchain modules
- Added relevant schemas

### How was it tested?

- Added unit tests

### CODE REVIEW

File: framework/src/modules/interoperability/mainchain/endpoint.ts [hunk 1]
```diff
@@ -13,5 +13,51 @@
...
+	protected readonly interoperableCCAPIs = new Map<number, BaseInteroperableAPI>();
+
+	public constructor(moduleID: Buffer, interoperableCCAPIs: Map<number, BaseInteroperableAPI>) {
+		super(moduleID);
+		this.interoperableCCAPIs = interoperableCCAPIs;
+	}
+
+	public async getChainAccount(context: ModuleEndpointContext, chainID: Buffer) {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		await interoperabilityStore.getChainAccount(chainID);
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/endpoint.ts:31]: Missing `return` here and on other places, I missed this in the last review for API but its fixed now

File: framework/src/modules/interoperability/mainchain/module.ts [hunk 2]
```diff
@@ -18,15 +18,55 @@ import { BaseInteroperabilityModule } from '../base_interoperability_module';
...
+	terminatedOutboxSchema,
+} from '../schemas';
 
 export class MainchainInteroperabilityModule extends BaseInteroperabilityModule {
 	public crossChainAPI = new MainchainCCAPI(this.id);
 	public api = new MainchainInteroperabilityAPI(this.id, this.interoperableCCAPIs);
-	public endpoint = new MainchainInteroperabilityEndpoint(this.id);
+	public endpoint = new MainchainInteroperabilityEndpoint(this.id, this.interoperableCCAPIs);
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	public registerInteroperableModule(): void {
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/module.ts:38]: why add this as its already implemented in the base class?

File: framework/src/modules/interoperability/schemas.ts [hunk 3]
```diff
@@ -746,3 +746,33 @@ export const stateRecoveryInitParams = {
...
+	properties: {
+		chainID: {
+			dataType: 'bytes',
+			fieldNumber: 1,
+		},
+	},
+};
+
+export const getChainAccountResponseSchema = {
+	$id: '/modules/interoperability/endpoint/getChainAccountRequest',
```
Comment (ishantiw) [inline framework/src/modules/interoperability/schemas.ts:763]:
```suggestion
	$id: '/modules/interoperability/endpoint/getChainAccountResponse',
```

File: framework/src/modules/interoperability/sidechain/endpoint.ts [hunk 4]
```diff
@@ -13,5 +13,51 @@
...
+	protected readonly interoperableCCAPIs = new Map<number, BaseInteroperableAPI>();
+
+	public constructor(moduleID: Buffer, interoperableCCAPIs: Map<number, BaseInteroperableAPI>) {
+		super(moduleID);
+		this.interoperableCCAPIs = interoperableCCAPIs;
+	}
+
+	public async getChainAccount(context: ModuleEndpointContext, chainID: Buffer) {
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		await interoperabilityStore.getChainAccount(chainID);
```
Comment (ishantiw) [inline framework/src/modules/interoperability/sidechain/endpoint.ts:31]: missing `return` and also in other places

**Review by ishantiw — APPROVED** [review]

**Review by Incede — APPROVED** [review]

**Pull request merged**